includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -513,6 +513,74 @@ public function sanitize() {
 		}
 	}
 
+	/**
+	 * Eliminate relative segments (../ and ./) from a path.
+	 *
+	 * @param string $path Path with relative segments. This is not a URL, so no host and no query string.
+	 * @return string|WP_Error Unrelativized path or WP_Error if there is too much relativity.
+	 */
+	private function unrelativize_path( $path ) {
+		// Eliminate current directory relative paths, like <foo/./bar/./baz.css> => <foo/bar/baz.css>.
+		do {
+			$path = preg_replace(
+				'#/\./#',
+				'/',
+				$path,
+				-1,
+				$count
+			);
+		} while ( 0 !== $count );
+
+		// Collapse relative paths, like <foo/bar/../../baz.css> => <baz.css>.
+		do {
+			$path = preg_replace(
+				'#(?<=/)(?!\.\./)[^/]+/\.\./#',
+				'',
+				$path,
+				1,
+				$count
+			);
+		} while ( 0 !== $count );
+
+		if ( false !== strpos( $path, './' ) ) {
```

Comment:
Is this accurate enough? Just wondering if someone named their directories weird (`mypath/my-dir-with-dot./something.css`)... edge-case, but should we rather use regex here?

Reply:
How about:

```php
if ( preg_match( '#/\.+/#', $path ) ) {
```

Reply:
Actually, I think we can remove this condition entirely as it is overkill. It shouldn't happen because of the first do/while loop. And if it fails then it will be caught below in:

https://github.com/ampproject/amp-wp/blob/9473cbd9aeb1af1ac0c3191895e3456520c4b161/includes/sanitizers/class-amp-style-sanitizer.php#L674-L677

Reply:
Meh, let's keep it as `unrelativize_path` is called in another place.